### DISCUSSION TITLE
Selective Sync Conflict

### DISCUSSION THREAD

**Discussion by ghost** [samschott/maestral#749] [Q&A] [answered]
I messed up a bit.

Basically I wanted to change secret service from gnome-keyring to keepassxc.

I ended up removing gnome-keyring, then maestral couldn't find the keys anymore, so I ended up uninstalling maestral, removing ~/.config/maestral, ~/.cache, ~/.local/maestral, and reinstalling maestral again.

Re-linking my account, select as sync folder the same sync folder that were there before, and accepted to merge the folders.

now I have some folder with "(selective sync)" and also double "(selective sync) (selective sync").

What can I do?

Can I rename those folders to remove the "(selective sync)" from their name?

**Comment by ghost** [comment]
Any help? :(

**Comment by ghost** [comment]
I removed the "selective sync" folders from the dropbox website and maestral is syncing that removal on local folder.
I don't know if that is the correct way to do it but i have to move on.

**Comment by samschott** [comment, marked as answer]
Apologies for not seeing this earlier. Removing the "selective sync" folders should be safe. They will be created if an item that exists or was created locally matches the name of an item excluded by selective sync.

I assume that you had previously downloaded folders which you, during the new setup, excluded from syncing. Instead of deleting the local folder during the initial sync, Maestral will create a "selective sync" conflict. Those will be safe to delete unless you made any changes to the local items in the meantime which are not yet reflected in the original folder.

**Comment by ghost** [comment]
> They will be created if an item that exists or was created locally matches the name of an item excluded by selective sync.

So in case I reinstall maestral and don't want it to re-sync everything, I should use the old local Dropbox folder, and in "selective sync" I should select the same exact folders I have in the (old, now current) local Dropbox folder?

So my understanding is that if I don't, the local folder which has the same name a remote folder, gets renamed, and a "(selective sync conflict)" is appended to its name?

> I assume that you had previously downloaded folders which you, during the new setup, excluded from syncing. Instead of deleting the local folder during the initial sync, Maestral will create a "selective sync" conflict. Those will be safe to delete unless you made any changes to the local items in the meantime which are not yet reflected in the original folder.

And that "new" selective sync folders, which, if I didn't make any modifications in the meantime, are the same as the remote folders, gets uploaded in the remote?

So the end result is that both remote and local contain the "selective sync" folder, but only the remote contain the "original" folder (with no selective sync added to it)?

**Comment by samschott** [comment]
Yes, your understanding is correct in all points. The reason for this is that Maestral does not know your intentions when you have excluded a remote folder but then create a local one with the same name. In principle your intent could be:

* You want merge their contents and automatically include this folder again. Sync conflicts should be resolved automatically if possible.
* You forgot about the folder in the cloud and are trying to create a different one locally.
* You are merging with a local folder structure and want to now delete the local folders which are excluded.

Rather than trying to guess, or presenting you with a dialog to choose, Maestral will bail out and rename the local folder by appending "selective sync conflict" so that none of its content is lost.
